app/controllers/users_controller.rb

```diff
@@ -18,10 +22,59 @@ def create
     end
   end
 
+  def edit; end
+
+  def index
+    @pagy, @users = pagy User.all, items: Settings.page_30
+  end
+
+  def update
+    if @user.update user_params
+      flash[:update_sucess] = t("flash.update_succ")
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    if @user.destroy
+      flash[:success] = t("flash.user_delete")
+      redirect_to users_url, status: :see_other
+    else
+      flash[:danger] = t("flash.user_delete_fail")
+      redirect_to users_url, status: :unprocessable_entity
+    end
+  end
+
+  def logged_in_user
+    return if logged_in?
+
+    flash[:danger] = t("flash.logged_in")
+    store_location
+    redirect_to login_url, status: :see_other
+  end
+
   private
 
   def user_params
     params.require(:user).permit :name, :email, :password,
                                  :password_confirmation
   end
+
+  def find_user
+    @user = User.find_by id: params[:id]
+    redirect_to root_path unless @user
```

Comment:
```suggestion
    return if @user
    flash[:error] = "User not found" # I18n nhé
    redirect_to root_path
```
ri hì, thân thiện ng dùng nào em